front50-core/src/main/java/com/netflix/spinnaker/front50/validator/AppNameValidator.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.front50.validator;
+
+import com.google.common.base.Strings;
+import com.netflix.spinnaker.front50.model.application.Application;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AppNameValidator implements ApplicationValidator {
+  private String validationRegEx;
+
+  @Autowired
+  public AppNameValidator(
+      @Value("${validation.app-name-validator.name-reg-ex:#{null}}}") String validationRegEx) {
```

Comment:
I know its a bit of a pain in the ass, but if this can be in a ConfigurationProperties rather than an Value it sets us up to better document configuration

Reply:
you're right, i was being lazy... will fix!